### DISCUSSION TITLE
Can't filter based on relational data

### DISCUSSION THREAD

**Discussion by tmrkp** [directus/directus#2838] [Archive]
I want to filter the data included by a `o2m` field.

This is my request: `/api/items/pages?fields=blocks.level&filter[blocks.level][eq]=1`.

`pages` is a collection with a `o2m` field that is related to the collection `blocks`.

But I get data in the response that shouldn't be included:

```
{
  "data": [
    {
      "blocks": [
        {
          "level": 1
        }
      ]
    },
    {
      "blocks": [
        {
          "level": 1
        },
        {                <-
          "level": 2     <- i'd like to exclude this
        }                <-
      ]
    }
  ]
}
```

Am I missing something or is the `o2m` field just not capable of filtering?

**Comment by rijkvanzanten** [comment]
O2Ms in v8 don't offer this level of filtering.

`&filter[blocks.level][eq]=1` filters the top level items where the top level item has a nested child that has level=1

**Reply by rijkvanzanten** [reply]
v9 (coming soon, see [directus/next](https://github.com/directus/next)) will offer a "deep" query param that allows you to add any of the query params to the nested set. To achieve your example, you'd be able to do `?deep[blocks][filter][level][_eq]=1`

**Reply by tmrkp** [reply]
Thanks for your reply!

For now, I will split the request into two separate requests.
